src/test/fuzz/txdownloadman_one_honest_peer.cpp

```diff
@@ -0,0 +1,436 @@
+// send a malleated or witness stripped tx/inv, send a notfound, etc.
+FUZZ_TARGET(tx_download_one_honest_peer, .init = initialize)
+{
+    FuzzedDataProvider fuzzed_data_provider(buffer.data(), buffer.size());
+
+    // Initialize txdownloadman
+    bilingual_str error;
+    CTxMemPool pool{MemPoolOptionsForTest(g_setup->m_node), error};
+    const auto max_orphan_count = fuzzed_data_provider.ConsumeIntegralInRange<unsigned int>(0, 300);
+    FastRandomContext det_rand{true};
+
+    // Make target transaction, malleated, and stripped
+    auto target_tx{MakeTransactionSpending(
+        {fuzzed_data_provider.PickValueInArray(COINS)},
+        /*num_outputs=*/2,
+        /*add_witness=*/fuzzed_data_provider.ConsumeBool())
+    };
+    auto malleated_tx = Malleate(target_tx);
+    auto stripped_tx = StripWitness(target_tx);
+    const auto& target_txid = target_tx->GetHash().ToUint256();
+    const auto& target_wtxid = target_tx->GetWitnessHash().ToUint256();
+    const auto& malleated_wtxid = malleated_tx->GetWitnessHash().ToUint256();
+
+    node::TxDownloadManager txdownloadman{node::TxDownloadOptions{pool, det_rand, max_orphan_count}};
+    std::chrono::microseconds time{244466666};
+
+    // Decide what the peers are.
+    std::vector<TestPeer> all_peers;
+    // NodeId of the "designated good peer"
+    // Other TestPeers may be "good" as well, but not guaranteed.
+    NodeId honestid = fuzzed_data_provider.ConsumeIntegralInRange<unsigned int>(0, 7);
+    // 8 outbound peers. NodeId = o. We indicate outbound using preferred=true
+    for (unsigned int o = 0; o < 8; ++o) {
+        bool good = (o == honestid) || fuzzed_data_provider.ConsumeBool();
+        all_peers.emplace_back(RandomPeer(fuzzed_data_provider, o, /*preferred=*/true, /*force_good=*/good));
+    }
+    // 0-120 inbound peers. NodeId = i. We indicate inbound using preferred=false
+    auto num_inbound_peers = fuzzed_data_provider.ConsumeIntegralInRange<unsigned int>(0, 120);
+    for (unsigned int i = 8; i < num_inbound_peers + 8; ++i) {
+        bool good = fuzzed_data_provider.ConsumeBool();
+        all_peers.emplace_back(RandomPeer(fuzzed_data_provider, i, /*preferred=*/false, /*force_good=*/good));
+    }
+
+    // Indexes for random peer iteration
+    std::vector<size_t> indexes;
+    indexes.resize(all_peers.size());
+    std::iota(indexes.begin(), indexes.end(), 0);
+
+    // We are trying to ensure that we always download the tx as long as we have 1 good outbound
+    // peer who is able to send it to us.
+    bool got_tx{false};
+
+    // Keep track of how many iterations the loop has done to make sure the target transaction is
+    // announced within the first 1000 rounds. Note that the good peer is guaranteed to connect
+    // within the first round and will never be disconnected.
+    int counter = 0;
+
+    LIMITED_WHILE(!got_tx, 100000)
+    {
+        std::shuffle(indexes.begin(), indexes.end(), det_rand);
```

Comment:
Any reason not to let the fuzzer decide the order?

If the order is interesting (i.e. a specific order might trigger a bug) then I think it'd make more sense to let the fuzzer choose the order.